elkm1_lib/thermostats.py

```diff
@@ -3,31 +3,61 @@
 from __future__ import annotations
 
 from .connection import Connection
-from .const import Max, TextDescriptions
+from .const import (
+    Max,
+    TextDescriptions,
+    ThermostatFan,
+    ThermostatMode,
+    ThermostatSetting,
+)
 from .elements import Element, Elements
 from .message import tr_encode, ts_encode
 from .notify import Notifier
 
+SETTING_TYPING = {
+    ThermostatSetting.MODE: ThermostatMode,
+    ThermostatSetting.HOLD: bool,
+    ThermostatSetting.FAN: ThermostatFan,
+    ThermostatSetting.GET_TEMPERATURE: int,
+    ThermostatSetting.COOL_SETPOINT: int,
+    ThermostatSetting.HEAT_SETPOINT: int,
+}
+
 
 class Thermostat(Element):
     """Class representing an Thermostat"""
 
     def __init__(self, index: int, connection: Connection, notifier: Notifier) -> None:
         super().__init__(index, connection, notifier)
-        self.mode = 0
+        self.mode: ThermostatMode | None = None
         self.hold = False
-        self.fan = 0
+        self.fan: ThermostatFan | None = None
         self.current_temp = 0
         self.heat_setpoint = 0
         self.cool_setpoint = 0
         self.humidity = 0
 
-    def set(self, element_to_set: int, value: int) -> None:
+    def set(
+        self,
+        element_to_set: ThermostatSetting,
+        val: bool | int | ThermostatMode | ThermostatFan,
+    ) -> None:
         """(Helper) Set thermostat"""
-        self._connection.send(ts_encode(self.index, value, element_to_set))
+        if (  # pylint: disable=unidiomatic-typecheck
+            type(val) != SETTING_TYPING[element_to_set]
+        ):
```

Comment:
Can you use `isinstance` here?

Reply:
Unfortunately no. `isinstance(x, int)` is true for both ints and bools.

Reply:
```
>>> int.__subclasses__()
[<class 'bool'>, <enum 'IntEnum'>, <enum 'IntFlag'>, <class 'sre_constants._NamedIntConstant'>]
```

Yeah I guess you would need a special case that checks to make sure its `int`s is `not isinstance(val, bool)`